### PR TITLE
Use malloc/memset not calloc for WinCE portability

### PR DESCRIPTION
**Pull request by richsalz**
Fixes: #2539

### CODE REVIEW

File: crypto/LPdir_win.c [hunk 1]
```diff
@@ -105,6 +106,7 @@ const char *LP_find_file(LP_DIR_CTX **ctx, const char *directory)
                 errno = ENOMEM;
                 return 0;
             }
+            memset(wdir, 0, amount);
```
Comment (levitte) [inline crypto/LPdir_win.c]: This isn't really needed.

Comment (richsalz) [reply crypto/LPdir_win.c]: I didn't think so, but the original code used ```calloc``` so I kept the semantics.

Comment (richsalz) [reply crypto/LPdir_win.c]: updated commit pushed to remove it; please re-review.

**Comment by levitte** [comment]
So here's a thought: as I recall it, the reason to use `calloc` was if the function was called with an extremely long string and the calculation `len_0 * sizeof(TCHAR)` overflows and results in a too small result in `amount`.  If that happens, we're allocating a too small space and will end up with a buffer overflow that is likely to trash heap info and possibly have the application crash horribly.

So I think it would be wise to check that `len_0 <= SIZE_MAX / sizeof(TCHAR)` (make sure to include `"internal/numbers.h"`

**Comment by richsalz** [comment]
Really?  The windows max pathlen is 260 chars (do a search).

**Comment by levitte** [comment]
This is a public function, we might as well assume it might be called maliciously.

**Comment by t-j-h** [comment]
I would leave the memset there ...

**Comment by levitte** [comment]
@t-j-h, why?

**Comment by richsalz** [comment]
I give up.  Sorry, @gjcoram.

**Comment by levitte** [comment]
Err, I'm sad to see that you give up, that was not my intention.

**Comment by richsalz** [comment]
The original fix was appropriate for the issue.  Hopefully someone else will try, my time ran out.

**Comment by levitte** [comment]
Okie...  would you mind having a look at #4594?

**Pull request closed without merging**